### PR TITLE
add feign client to api-gateway

### PR DESCRIPTION
**Pull request by secret-lain**

### CODE REVIEW

**Comment by mszarlinski** [comment]
Thanks for your contribution, please answer my comments. @arey could you look on it?

**Comment by secret-lain** [comment]
I do not think you should replace it.
But feign client is more abstraction than RestTemplate and I think it is more readable structure.
If netflix opensource such as hystrix is added in the future, I think it will be more flexible.

File: spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/application/CustomersServiceFeignClient.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.springframework.samples.petclinic.api.application;
+
+import org.springframework.cloud.netflix.feign.FeignClient;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+/**
+ * @author lain
+ */
+@FeignClient("customers-service")
```
[the remark this answers is not in the document]
Comment (Dovchiproeng) [reply spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/application/CustomersServiceFeignClient.java]: [https://cloud.spring.io/spring-cloud-netflix/multi/multi_spring-cloud-feign.html](Spring cloud document) did state this
> In the @FeignClient annotation the String value ("stores" above) is an arbitrary client name, which is used to create a Ribbon load balancer (see below for details of Ribbon support). You can also specify a URL using the url attribute (absolute value or just a hostname). The name of the bean in the application context is the fully qualified name of the interface. To specify your own alias value you can use the qualifier value of the @FeignClient annotation.

Hope it help with your question.

**Comment by arey** [comment]
Conflicting file comes from the PR #114. Hystrix circuit breaker has been added with a fallback if the `visits-service` is unavailable.

**Comment by arey** [comment]
@secret-lain we are now using the webflux WebClient for chaining call in a reactive way

**Pull request closed without merging**